Uplift/ViewModels/FitnessCenterViewModel.swift

```diff
@@ -45,10 +55,117 @@ extension FitnessCenterView {
                 if day == Date.now.getDayOfWeek() {
                     daysOfWeek.append("Today")
                 } else {
-                    daysOfWeek.append(day.dayOfWeekAbbreviation())
+                    daysOfWeek.append(day.dayOfWeekShortened())
                 }
             }
         }
-    }
 
+        /// Fetch hourly average capacities for a specific facility, filtered by its hours of operation.
+        func fetchHourlyAverageCapacities(for fc: Facility) {
+            popularTimesIsAnimating = false
+
+            guard let facilityId = Int(fc.id) else {
+                Logger.data.critical("Invalid facility ID")
+                return
+            }
+
+            Network.client.queryPublisher(
+                query: GetHourlyAverageCapacitiesByFacilityIdQuery(facilityId: facilityId),
+                cachePolicy: .fetchIgnoringCacheCompletely
+            )
+            .map { result -> [HourlyAverageCapacity] in
+                let hourlyAverage = result.data?.getHourlyAverageCapacitiesByFacilityId ?? []
+                let capacities = hourlyAverage.compactMap { cap -> HourlyAverageCapacity? in
+                    guard let cap,
+                          let dayOfWeek = cap.dayOfWeek?.rawValue else { return nil }
+
+                    if facilityId == cap.facilityId {
+                        return HourlyAverageCapacity(
+                            averagePercent: cap.averagePercent,
+                            dayOfWeek: dayOfWeek,
+                            hourOfDay: cap.hourOfDay
+                        )
+                    }
+
+                    return nil
```

Comment:
Why not add that condition to the guard
```suggestion
                    guard let cap,
                          let dayOfWeek = cap.dayOfWeek?.rawValue,
                            facilityId == cap.facilityId else { return nil }

                        return HourlyAverageCapacity(
                            averagePercent: cap.averagePercent,
                            dayOfWeek: dayOfWeek,
                            hourOfDay: cap.hourOfDay
                        )